pcsx2/Patch.cpp

```diff
@@ -901,6 +913,91 @@ void Patch::PatchFunc::gsinterlacemode(PatchGroup* group, const std::string_view
 	group->override_interlace_mode = static_cast<GSInterlaceMode>(interlace_mode.value());
 }
 
+void Patch::PatchFunc::dpatch(PatchGroup* group, const std::string_view cmd, const std::string_view param)
+{
+#define PATCH_ERROR(fstring, ...) \
+	Console.Error(fmt::format("(dPatch) Error Parsing: {}={}: " fstring, cmd, param, __VA_ARGS__))
+
+	// [0]=number of patterns,[1]=number of replacements
+	// Each pattern or replacement is [2]=offset,[3]=hex
+
+	const std::vector<std::string_view> pieces(StringUtil::SplitString(param, ',', false));
+	if (pieces.size() < 2)
+	{
+		PATCH_ERROR("Expected at least 2 data parameters; only found {}", pieces.size());
+		return;
+	}
+
+	std::string_view patterns_end, replacements_end;
+	std::optional<u32> num_patterns = StringUtil::FromChars<u32>(pieces[0], 16, &patterns_end);
+	std::optional<u32> num_replacements = StringUtil::FromChars<u32>(pieces[1], 16, &replacements_end);
+
+	if(!num_patterns.has_value())
+	{
+		PATCH_ERROR("Malformed number of patterns '{}', a decimal number is expected", pieces[0]);
+		return;
+	}
+
+	if(!num_replacements.has_value())
+	{
+		PATCH_ERROR("Malformed number of replacements '{}', a decimal number is expected", pieces[1]);
+		return;
+	}
+
+	if (pieces.size() != ((num_patterns.value() * 2) + (num_replacements.value() * 2) + 2))
+	{
+		PATCH_ERROR("Expected 2 fields for each {} patterns and {} replacements; only found {}", num_patterns.value(), num_replacements.value(), num_patterns.value() + num_replacements.value());
```

Comment:
The last one should be `"found {} fields", pieces.size() - 2`.

Without "only": there might be *more* pieces than expected.

Reply:
Good catch, thanks.